modules/ascat/main.nf

```diff
@@ -0,0 +1,155 @@
+        loci.prefix = "$loci_files/G1000_loci_${genomeVersion}_chr",
+        gender = "$gender",
+        genomeVersion = "$genomeVersion",
+        nthreads = $task.cpus
+        $minCounts_arg
+        $chrom_names_arg
+        $min_base_qual_arg
+        $min_map_qual_arg
+        $ref_fasta_arg
+        $skip_allele_counting_tumour_arg
+        $skip_allele_counting_normal_arg
+    )
+
+
+    #Load the data
+    ascat.bc = ascat.loadData(
+        Tumor_LogR_file = "Tumour_tumourLogR.txt",
+        Tumor_BAF_file = "Tumour_normalBAF.txt",
+        Germline_LogR_file = "Tumour_normalLogR.txt",
+        Germline_BAF_file = "Tumour_normalBAF.txt",
+        genomeVersion = "$genomeVersion",
+        gender = "$gender"
+    )
+
+    #optional GC wave correction
+    if(!is.null($gc_files)){
+        ascat.bc = ascat.GCcorrect(ascat.bc, $gc_files)
+    }
+
+    #Plot the raw data
+    ascat.plotRawData(ascat.bc)
+
+    #Segment the data
+    ascat.bc = ascat.aspcf(ascat.bc)
+
+    #Plot the segmented data
+    ascat.plotSegmentedData(ascat.bc)
+
+    #Run ASCAT to fit every tumor to a model, inferring ploidy, normal cell contamination, and discrete copy numbers
+    #If psi and rho are manually set:
+    if (!is.null($purity) && !is.null($ploidy)){
+        ascat.output <- ascat.runAscat(ascat.bc, gamma=1, rho_manual=$purity, psi_manual=$ploidy)
+    } else if(!is.null($purity) && is.null($ploidy)){
+        ascat.output <- ascat.runAscat(ascat.bc, gamma=1, rho_manual=$purity)
+    } else if(!is.null($ploidy) && is.null($purity)){
+        ascat.output <- ascat.runAscat(ascat.bc, gamma=1, psi_manual=$ploidy)
+    } else {
+        ascat.output <- ascat.runAscat(ascat.bc, gamma=1)
+    }
+
+    #Write out segmented regions (including regions with one copy of each allele)
+    write.table(ascat.output[["segments"]], file=paste0("$prefix", ".segments.txt"), sep="\t", quote=F, row.names=F)
+
+    #Write out CNVs in bed format
+    cnvs=ascat.output[["segments"]][2:6]
+    write.table(cnvs, file=paste0("$prefix",".cnvs.txt"), sep="\t", quote=F, row.names=F, col.names=T)
+
+    #Write out purity and ploidy info
+    summary <- tryCatch({
+            matrix(c(ascat.output[["aberrantcellfraction"]], ascat.output[["ploidy"]]), ncol=2, byrow=TRUE)}, error = function(err) {
+                # error handler picks up where error was generated
+                print(paste("Could not find optimal solution:  ",err))
+                return(matrix(c(0,0),nrow=1,ncol=2,byrow = TRUE))
+        }
+    )
+    colnames(summary) <- c("AberrantCellFraction","Ploidy")
+    write.table(summary, file=paste0("$prefix",".purityploidy.txt"), sep="\t", quote=F, row.names=F, col.names=T)
+
+    #version export. Have to hardcode process name and software name because
+    #won't run inside an R-block
+    version_file_path="versions.yml"
+    f <- file(version_file_path,"w")
+    writeLines("ASCAT:", f)
+    writeLines(" ascat: 3.0.0",f)
```

Comment:
No, I can't do that. I had the same trouble with [this module](https://github.com/nf-core/modules/blob/79a9d5e1eae6cc6f3b2a0a0d02e61ae4b5ef3748/modules/imputeme/vcftoprs/main.nf#L42-L48). Because that script block is running R-code, so it fails the 'usual' way of generating the versions.yml. I tried also with wrapping it into R-`system` calls to the bash code and such, but that looks even more hideous.

Reply:
ok makes sense, yes I forgot this is an R block